tsl/src/compression/create.c

```diff
@@ -923,24 +923,40 @@ disable_compression(Hypertable *ht, WithClauseResult *with_clause_options)
 }
 
 /* Add column to internal compression table */
-static void
+static bool
 add_column_to_compression_table(Hypertable *compress_ht, CompressColInfo *compress_cols)
 {
 	Oid compress_relid = compress_ht->main_table_relid;
 	ColumnDef *coldef;
 	AlterTableCmd *addcol_cmd;
 	coldef = (ColumnDef *) linitial(compress_cols->coldeflist);
+	HeapTuple attTuple;
+	bool new_column = false;
 
 	/* create altertable stmt to add column to the compressed hypertable */
 	Assert(TS_HYPERTABLE_IS_INTERNAL_COMPRESSION_TABLE(compress_ht));
 	addcol_cmd = makeNode(AlterTableCmd);
 	addcol_cmd->subtype = AT_AddColumn;
 	addcol_cmd->def = (Node *) coldef;
-	addcol_cmd->missing_ok = false;
+	addcol_cmd->missing_ok = true;
+
+	attTuple = SearchSysCache2(ATTNAME,
+							   ObjectIdGetDatum(compress_relid),
+							   PointerGetDatum(coldef->colname));
+	if (!HeapTupleIsValid(attTuple))
+	{
+		/* the column doesn't exist yet */
+		new_column = true;
+	}
+	else
```

Comment:
why is Release conditional? Shouldn't this be released before leaving the function?

Reply:
It is not needed when the search fails.